CONTRIBUTING.md

```diff
@@ -0,0 +1,94 @@
+
+There are many ways to contribute,
+from writing tutorials or blog posts,
+improving the documentation,
+submitting bug reports and feature requests or writing code.
+
+You can get in touch with us through [Discuss](https://discuss.elastic.co/c/apm),
+feedback and ideas are always welcome.
+
+Please note that this repository is covered by the [Elastic Community Code of Conduct](https://www.elastic.co/community/codeofconduct).
+
+## Code contributions
+
+If you have a bugfix or new feature that you would like to contribute,
+please find or open an issue about it first.
+Talk about what you would like to do.
+It may be that somebody is already working on it,
+or that there are particular issues that you should know about before implementing the change.
+
+### Submitting your changes
+
+Generally, we require that you test any code you are adding or modifying.
+Once your changes are ready to submit for review:
+
+1. Sign the Contributor License Agreement
+
+    Please make sure you have signed our [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
+    We are not asking you to assign copyright to us,
+    but to give us the right to distribute your code without restriction.
+    We ask this of all contributors in order to assure our users of the origin and continuing existence of the code.
+    You only need to sign the CLA once.
+
+2. Test your changes
+
+    Run the test suite to make sure that nothing is broken.
+    See [testing](#testing) for details.
+
+3. Rebase your changes
+
+    Update your local repository with the most recent code from the main repo,
+    and rebase your branch on top of the latest master branch.
+    We prefer your initial changes to be squashed into a single commit.
+    Later,
+    if we ask you to make changes,
+    add them as separate commits.
+    This makes them easier to review.
+    As a final step before merging we will either ask you to squash all commits yourself or we'll do it for you.
+
+4. Submit a pull request
+
+    Push your local changes to your forked copy of the repository and [submit a pull request](https://help.github.com/articles/using-pull-requests).
+    In the pull request,
+    choose a title which sums up the changes that you have made,
+    and in the body provide more details about what your changes do.
+    Also mention the number of the issue where discussion has taken place,
+    eg "Closes #123".
+
+5. Be patient
+
+    We might not be able to review your code as fast as we would like to,
+    but we'll do our best to dedicate it the attention it deserves.
+    Your effort is much appreciated!
+
+### Testing
+
+To run the full test suite,
+you need to install several databases (Elasticsearch, PostgreSQL, MySQL, Cassandra, Redis).
+This can be quite a hassle, so we recommend to use our dockerized test setup.
+See [Running tests](https://www.elastic.co/guide/en/apm/agent/python/current/run-tests-locally.html) for detailed instructions.
```

Comment:
Consider if you want to use `master` instead of `current` as people mostly will make PR's against the master branch

Reply:
good point!

Reply:
You can change the default target branch for PRs. For example, [npm](https://github.com/npm/npm) uses `latest` rather than `master` and actually deleted the `master` branch to avoid the potentially problematic terminology.

Reply:
@Qard I don't think a PR in this tiny little repo is the right venue for this discussion. If you think the usage of `master` in the context of git branches is problematic, I suggest to bring it up in a company/community wide forum.

Reply:
Oh, I didn't intend to put attention on that part. I was just trying to point out you can *change* the default target branch for PRs easily, regardless of whatever reason one might have to do that. I just provided that as an example. It was intended as a counter to Thomas stating that most people will PR against master. 😅